native/Makefile

```diff
@@ -26,10 +27,19 @@ all: $(addprefix bin/,$(BENCHMARKS))
 bin:
 	mkdir -p bin
 
+bin/log_reg_lbfgs: log_reg_lbfgs_bench.cpp $(FOBJ) | bin
+	$(CXX) $^ $(CXXINCLUDE) $(CXXFLAGS) $(LDFLAGS) -lmkl_rt -lm -lifcore \
```

Comment:
Linking against `mkl_rt`, are we making sure that MKL is using TBB as the underlying threading layer?

MKL's default is to use OpenMP, as since DAAL's default is to use TBB we end up incurring the cost of runtime of both.

One can either set the threading layer in the benchmark itself, by calling [`mkl_set_threading_layer`](https://software.intel.com/en-us/mkl-developer-reference-c-mkl-set-threading-layer), or use explicit dynamic linking, see [MKL Linking Advisor](https://software.intel.com/en-us/articles/intel-mkl-link-line-advisor).

Reply:
I added a call to `mkl_set_threading_layer` in `native/log_reg_lbfgs_bench.cpp`.